### ISSUE TITLE
Your `export_method` in gym is defined as `development`

### ISSUE THREAD

**Issue by jamshaidishaque** [fastlane/fastlane#16002]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

```
 PhaseScriptExecution [CP]\ Embed\ Pods\ Frameworks /Users/build/Library/Developer/Xcode/DerivedData/John-eyhufzbqcmuyuhahrcsrieyluqyl/Build/Intermediates.noindex/ArchiveIntermediates/Johnpay/IntermediateBuildFilesPath/Johnpay.build/Debug-iphoneos/Johnpay.build/Script-BDAC8382C754A82925A108E1.sh
(1 failure)
```

##### Command executed

fastlane build_ios_debug

##### Complete output when running fastlane, including the stack trace and command used 

```
▸ Compiling LaunchScreen.storyboard
▸ Processing Info.plist
▸ Running script '[CP] Embed Pods Frameworks'
▸ Generating 'Johnpay.app.dSYM'
** ARCHIVE FAILED **


The following build commands failed:
        PhaseScriptExecution [CP]\ Embed\ Pods\ Frameworks /Users/build/Library/Developer/Xcode/DerivedData/John-eyhufzbqcmuyuhahrcsrieyluqyl/Build/Intermediates.noindex/ArchiveIntermediates/Johnpay/IntermediateBuildFilesPath/Johnpay.build/Debug-iphoneos/Johnpay.build/Script-BDAC8382C754A82925A108E1.sh
(1 failure)
[12:47:51]: Exit status: 65

+---------------+--------------------------------------------+
|                     Build environment                      |
+---------------+--------------------------------------------+
| xcode_path    | /Applications/XCode10.2.1/Xcode_10.2.1.app |
| gym_version   | 2.141.0                                    |
| export_method | development                                |
| sdk           | iPhoneOS12.2.sdk                           |
+---------------+--------------------------------------------+

[12:47:51]: ▸ Command PhaseScriptExecution failed with a nonzero exit code
[12:47:51]: ▸ GenerateDSYMFile /Users/build/Library/Developer/Xcode/DerivedData/Johnpay-eyhufzbqcmuyuhahrcsrieyluqyl/Build/Intermediates.noindex/ArchiveIntermediates/Johnpay/BuildProductsPath/Debug-iphoneos/Johnpay.app.dSYM /Users/build/Library/Developer/Xcode/DerivedData/Johnpay-eyhufzbqcmuyuhahrcsrieyluqyl/Build/Intermediates.noindex/ArchiveIntermediates/Johnpay/InstallationBuildProductsLocation/Applications/Johnpay.app/Johnpay (in target: Johnpay)
[12:47:51]: ▸     cd /Users/build/Desktop/jenkins/workspace/workspace/ios-app/Johnpay
[12:47:51]: ▸     /Applications/XCode10.2.1/Xcode_10.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dsymutil /Users/build/Library/Developer/Xcode/DerivedData/Johnpay-eyhufzbqcmuyuhahrcsrieyluqyl/Build/Intermediates.noindex/ArchiveIntermediates/Johnpay/InstallationBuildProductsLocation/Applications/Johnpay.app/Johnpay -o /Users/build/Library/Developer/Xcode/DerivedData/Johnpay-eyhufzbqcmuyuhahrcsrieyluqyl/Build/Intermediates.noindex/ArchiveIntermediates/Johnpay/BuildProductsPath/Debug-iphoneos/Johnpay.app.dSYM
[12:47:51]:
[12:47:51]: ⬆️  Check out the few lines of raw `xcodebuild` output above for potential hints on how to solve this error
[12:47:51]: 📋  For the complete and more detailed error log, check the full log at:
[12:47:51]: 📋  /Users/build/Library/Logs/gym/Johnpay-Johnpay.log
[12:47:51]:
[12:47:51]: Your `export_method` in gym is defined as `development`
[12:47:51]: which might cause problems when signing your application
[12:47:51]: Are you sure want to build and export for development?
[12:47:51]: Please make sure to define the correct export methods when calling
[12:47:51]: gym in your Fastfile or from the command line
[12:47:51]:
[12:47:51]:
[12:47:51]: Looks like fastlane ran into a build/archive error with your project
[12:47:51]: It's hard to tell what's causing the error, so we wrote some guides on how
[12:47:51]: to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
[12:47:51]: Before submitting an issue on GitHub, please follow the guide above and make
[12:47:51]: sure your project is set up correctly.
[12:47:51]: fastlane uses `xcodebuild` commands to generate your binary, you can see the
[12:47:51]: the full commands printed out in yellow in the above log.
[12:47:51]: Make sure to inspect the output above, as usually you'll find more error information there
[12:47:51]:
+------------------+-----------------+
|            Lane Context            |
+------------------+-----------------+
| DEFAULT_PLATFORM | ios             |
| PLATFORM_NAME    |                 |
| LANE_NAME        | build_ios_debug |
+------------------+-----------------+
[12:47:51]: Error building the application - see the log above

+------+------------------+-------------+
|           fastlane summary            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 💥   | gym              | 107         |
+------+------------------+-------------+

[12:47:51]: fastlane finished with errors

[!] Error building the application - see the log above
```

### Environment

[12:59:39]: fastlane detected a Gemfile in the current directory
[12:59:39]: However, it seems like you didn't use `bundle exec`
[12:59:39]: To launch fastlane faster, please use
[12:59:39]:
[12:59:39]: $ bundle exec fastlane env
[12:59:39]:
[12:59:39]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[12:59:41]: Generating fastlane environment output, this might take a few seconds...
Your Gemfile lists the gem fastlane-plugin-firebase_app_distribution (>= 0) more than once.
You should probably keep only one of them.
Remove any duplicate entries and specify the gem only once.
While it's not a problem now, it could cause errors if you change the version of one of them later.
[12:59:41]: Plugin 'FirebaseAppDistribution' overwrites already loaded action 'firebase_app_distribution'
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                             |
| --------------------------- | ----------------------------------------------------------------- |
| OS                          | 10.14.6                                                           |
| Ruby                        | 2.6.5                                                             |
| Bundler?                    | false                                                             |
| Git                         | git version 2.24.0                                                |
| Installation Source         | /usr/local/lib/ruby/gems/2.6.0/gems/fastlane-2.141.0/bin/fastlane |
| Host                        | Mac OS X 10.14.6 (18G103)                                         |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.6.5/lib                                  |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                                       |
| Is contained                | false                                                             |
| Is homebrew                 | false                                                             |
| Is installed via Fabric.app | false                                                             |
| Xcode Path                  | /Applications/XCode10.2.1/Xcode_10.2.1.app/Contents/Developer/    |
| Xcode Version               | 10.2.1                                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version


default_platform :ios


  desc "Submit a new Beta Build to Apple TestFlight"
  lane :build_ios_debug do
    # match(type: "appstore") # more information: https://codesigning.guide
    gym(
      configuration: 'Debug',
      scheme: 'Johnpay',
      export_method: 'development',
      export_options: {
        'signingStyle' => 'manual',
        teamID: "MAP432T5M7",
        provisioningProfiles: {
        "co.john.app.release": "Release - john Development Profile"
        }
      },

      output_directory: "/Users/build/Desktop/jenkins/workspace/workspace/ios-app/Johnpay/output/IPA", # Destination directory. Defaults to current directory.
      output_name: "John_Release.ipa"
    ) # Build your app - more options available
  end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
#app_identifier "" # The bundle identifier of your app
#apple_id "" # Your Apple email address

#team_id "[[DEV_PORTAL_TEAM_ID]]" # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.141.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                    | Version | Update-Status |
| ----------------------------------------- | ------- | ------------- |
| fastlane-plugin-firebase_app_distribution | 0.1.4   | ✅ Up-To-Date  |
| fastlane-plugin-firebase_app_distribution | 0.1.4   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                       | Version      |
| ----------------------------------------- | ------------ |
| did_you_mean                              | 1.3.0        |
| slack-notifier                            | 2.3.2        |
| atomos                                    | 0.1.3        |
| CFPropertyList                            | 3.0.2        |
| claide                                    | 1.0.3        |
| colored2                                  | 3.1.2        |
| nanaimo                                   | 0.2.6        |
| xcodeproj                                 | 1.15.0       |
| rouge                                     | 2.0.7        |
| xcpretty                                  | 0.3.0        |
| terminal-notifier                         | 2.0.0        |
| unicode-display_width                     | 1.6.1        |
| terminal-table                            | 1.8.0        |
| plist                                     | 3.5.0        |
| public_suffix                             | 2.0.5        |
| addressable                               | 2.7.0        |
| multipart-post                            | 2.0.0        |
| word_wrap                                 | 1.0.0        |
| tty-screen                                | 0.7.1        |
| tty-cursor                                | 0.7.1        |
| tty-spinner                               | 0.9.3        |
| babosa                                    | 1.0.3        |
| colored                                   | 1.2          |
| highline                                  | 1.7.10       |
| commander-fastlane                        | 4.4.6        |
| excon                                     | 0.72.0       |
| faraday                                   | 0.17.3       |
| unf_ext                                   | 0.0.7.6      |
| unf                                       | 0.1.4        |
| domain_name                               | 0.5.20190701 |
| http-cookie                               | 1.0.3        |
| faraday-cookie_jar                        | 0.0.6        |
| faraday_middleware                        | 0.13.1       |
| fastimage                                 | 2.1.7        |
| gh_inspector                              | 1.1.3        |
| json                                      | 2.3.0        |
| mini_magick                               | 4.10.1       |
| multi_xml                                 | 0.6.0        |
| rubyzip                                   | 1.3.0        |
| security                                  | 0.1.3        |
| xcpretty-travis-formatter                 | 1.0.0        |
| dotenv                                    | 2.7.5        |
| bundler                                   | 2.1.2        |
| naturally                                 | 2.2.0        |
| simctl                                    | 1.6.7        |
| jwt                                       | 2.1.0        |
| uber                                      | 0.1.0        |
| declarative                               | 0.0.10       |
| declarative-option                        | 0.1.0        |
| representable                             | 3.0.4        |
| retriable                                 | 3.1.2        |
| mini_mime                                 | 1.0.2        |
| multi_json                                | 1.14.1       |
| signet                                    | 0.12.0       |
| memoist                                   | 0.16.2       |
| os                                        | 1.0.1        |
| googleauth                                | 0.10.0       |
| httpclient                                | 2.8.3        |
| google-api-client                         | 0.36.4       |
| google-cloud-env                          | 1.3.0        |
| google-cloud-errors                       | 1.0.0        |
| google-cloud-core                         | 1.5.0        |
| digest-crc                                | 0.4.1        |
| google-cloud-storage                      | 1.25.1       |
| emoji_regex                               | 1.0.1        |
| fastlane-plugin-firebase_app_distribution | 0.1.4        |
</details>

**Comment by janpio** [comment]
Did you look into:

```
The following build commands failed:
        PhaseScriptExecution [CP]\ Embed\ Pods\ Frameworks /Users/build/Library/Developer/Xcode/DerivedData/John-eyhufzbqcmuyuhahrcsrieyluqyl/Build/Intermediates.noindex/ArchiveIntermediates/Johnpay/IntermediateBuildFilesPath/Johnpay.build/Debug-iphoneos/Johnpay.build/Script-BDAC8382C754A82925A108E1.sh
(1 failure)
[12:47:51]: Exit status: 65

+---------------+--------------------------------------------+
|                     Build environment                      |
+---------------+--------------------------------------------+
| xcode_path    | /Applications/XCode10.2.1/Xcode_10.2.1.app |
| gym_version   | 2.141.0                                    |
| export_method | development                                |
| sdk           | iPhoneOS12.2.sdk                           |
+---------------+--------------------------------------------+

[12:47:51]: ▸ Command PhaseScriptExecution failed with a nonzero exit code
```
?

**Comment by jamshaidishaque** [comment]
When I run the command from the Mac OS terminal, it's completed successfully. When I am trying to execute it using Jenkins via SSH method, then I am getting the above error.

**Comment by jamshaidishaque** [comment]
After connecting the Mac OS via SSH run the following command before running the fastlane command.

security unlock-keychain -p <your-password>
